# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1236059100 18000
#      Tue Mar 03 00:45:00 2009 -0500
# Node ID 821f0242e8c153592b1232ca4fa3c0d3bb255aa6
# Parent  63ad1133d0eda7e0a10dacd3a53c1705dfa8980d
gnuplot_drawnow.m: Fix unintended shift of plot image for the gnuplot postscript+eps terminal.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-03-03  Ben Abbott <bpabbott@mac.com>
+
+	* plot/gnuplot_drawnow.m: Fix unintended shift of plot image for
+	the gnuplot postscript+eps terminal.
+
 2009-03-02  Jaroslav Hajek  <highegg@gmail.com>
 
 	* pkg/pkg.m (fix_depends): Fix & simplify splitting the string.
 
 2009-03-02  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
 	* set/complement.m, set/union.m: Remove reference to deprecated
 	function create_set.
diff --git a/scripts/plot/gnuplot_drawnow.m b/scripts/plot/gnuplot_drawnow.m
--- a/scripts/plot/gnuplot_drawnow.m
+++ b/scripts/plot/gnuplot_drawnow.m
@@ -71,18 +71,24 @@ function gnuplot_drawnow (h, term, file,
   else
     print_usage ();
   endif
 
 endfunction
 
 function implicit_margin = gnuplot_implicit_margin (term, opts_str)
   ## gnuplot has an implicit margin of 50pts for PS output.
-  if (strcmpi (term, "postscript") && isempty (strfind (opts_str, "eps")))
-    implicit_margin = 50/72;
+  if (strcmpi (term, "postscript"))
+    if (isempty (strfind (opts_str, " eps"))
+        && isempty (strfind (opts_str, "eps ")))
+      implicit_margin = 50/72;
+    else
+      ## When zero, the behavior of gnuplot changes.
+      implicit_margin = 1/72;
+    endif
   else
     implicit_margin = 0.0;
   endif
 endfunction
 
 function plot_stream = open_gnuplot_stream (npipes, h)
   cmd = gnuplot_binary ();
   if (npipes > 1)
